pkg/controller/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -1241,6 +1241,19 @@ func (r *ReconcileAWSPrivateLink) updatePrivateLinkStatus(cd *hivev1.ClusterDepl
 	})
 }
 
+func (r *ReconcileAWSPrivateLink) updateClusterDeploymentConditions(cd *hivev1.ClusterDeployment, logger log.FieldLogger) error {
+	return retry.RetryOnConflict(retryBackoff, func() error {
+		curr := &hivev1.ClusterDeployment{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: cd.Namespace, Name: cd.Name}, curr)
+		if err != nil {
+			return err
+		}
+
+		curr.Status.Conditions = cd.Status.Conditions
+		return r.Client.Status().Update(context.TODO(), curr)
+	})
+}
```

Comment:
Throughout hive I believe we always just return the error and let reconcile run again, is there any reason not to do the same here?

Reply:
i think it is common for controllers to use the retry on conflict when updating status. i personally would recommend every status update use it as it prevents noise in reconcile errors.

also the privatelinkcontroller reconcile has a little higher cost and error and reconcile again wouldn't be the best.

Reply:
That's reasonable, I agree we should try to do this more then.

cc @openshift/openshift-team-hive

Reply:
Won't this cause the controller to lose any changes made to conditions by other controllers?
1. Private Link controller wants to change condition A.
2. Some other controller changes condition B.
3. Private Link controller gets a conflict. Private Link controller overwrites all of the conditions, losing the changes to condition B.

Reply:
Yeah looks like you're right, need to surgically re-add the condition after fetching latest state.

Reply:
Ah, yes 🤦🏽